File: 2023-2024/sourcecode/mod4/lab12/ResourcePatternBasicsJava/app/src/main/java/com/example/resourcepatternbasicsjava/ui/CommentAdapter.java
package com.example.resourcepatternbasicsjava.ui;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.ListAdapter;

import com.example.resourcepatternbasicsjava.data.CommentResponseItem;
import com.example.resourcepatternbasicsjava.databinding.RvItemBinding;

public class CommentAdapter extends ListAdapter<CommentResponseItem, CommentViewHolder> {
    public CommentAdapter(CommentComparator userComparator) {
        super(userComparator);
    }

    @NonNull
    @Override
    public CommentViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        RvItemBinding binding = RvItemBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        return new CommentViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(CommentViewHolder holder, int position) {
        CommentResponseItem item = getItem(position);
        holder.bind(item);
    }
}

File: 2023-2024/sourcecode/mod3/ResourcePatternBasicsKotlin/app/src/main/java/com/example/resourcepatternbasicskotlin/ui/CommentAdapter.kt
package com.example.resourcepatternbasicskotlin.ui

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.ListAdapter
import com.example.resourcepatternbasicskotlin.data.CommentResponseItem
import com.example.resourcepatternbasicskotlin.databinding.RvItemBinding

class CommentAdapter(userComparator: CommentComparator) : ListAdapter<CommentResponseItem, CommentViewHolder>(userComparator) {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CommentViewHolder {
        return CommentViewHolder(
            RvItemBinding.inflate(
                LayoutInflater.from(parent.context), parent, false
            )
        )
    }

    override fun onBindViewHolder(holder: CommentViewHolder, position: Int) {
        val item = getItem(position)
        holder.bind(item)
    }
}